### PR TITLE
Install Che&Pipelines&Serverless operators to member cluster

### PR DESCRIPTION
**Pull request by alexeykazakov**
Previously the dev tools operators were installed manually in the crt member cluster. This PR automates the operator installations during cluster provisioning.
Later we will add more operators to our Toolchain Operator and switch to Toolchain Operator completely instead of installing Che, Pipelines, Serverless, etc individually.

### CODE REVIEW

File: config/operators/pipelines/subscription.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: openshift-pipelines-operator
+  namespace: openshift-operators
+spec:
+  channel: canary
+  installPlanApproval: Automatic
+  name: openshift-pipelines-operator
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: openshift-pipelines-operator.v0.11.0
```
Comment (MatousJobanek) [inline config/operators/pipelines/subscription.yaml]: we cannot use the latest OSP until we migrate all our pipelines to the `v1beta1`

File: config/operators/che/subscription.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: eclipse-che
+  namespace: toolchain-che
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: eclipse-che
```
Comment (MatousJobanek) [inline config/operators/che/subscription.yaml:24]: I thought that we cannot use the Che operator, but we need to rely on the Che cluster installed manually...

Comment (alexeykazakov) [reply config/operators/che/subscription.yaml:24]: What do you mean?
We use Che operator in Hosted Toolchain. Not CRW. And CheCluster is created by the script too: https://github.com/codeready-toolchain/toolchain-infra/pull/12/files#diff-a88c1d3771057755bf3be4f6e244165fR114

File: config/operators/pipelines/subscription.yaml (hunk 1 above)
Comment (alexeykazakov) [reply config/operators/pipelines/subscription.yaml]: Right. Let's stick with 0.10.7 for now - https://github.com/codeready-toolchain/toolchain-infra/pull/12/commits/20dcebea5e0ef5c72e5def50c139550c936b1afd

File: config/operators/che/subscription.yaml (hunk 2 above)
Comment (alexeykazakov) [reply config/operators/che/subscription.yaml:24]: Just for the clarification. Installing a Che operator is two step process:
- Install the Operator
- Create a CheCluster resource
This is what we did in our member cluster and this is exactly what this script is doing.

**Comment by xcoulon** [comment]
> _No description provided._

could you provide a description of the PR, please? 
for example, is this PR about automating the installation of Che (and Serverless) operators, while it was done manually in the past for Che?

**Comment by alexeykazakov** [comment]
> could you provide a description of the PR, please?
for example, is this PR about automating the installation of Che (and Serverless) operators, while it was done manually in the past for Che?

That’s correct. I thought the title explains it enough. Will add more details to the description.

**Comment by xcoulon** [comment]
> > could you provide a description of the PR, please?
> > for example, is this PR about automating the installation of Che (and Serverless) operators, while it was done manually in the past for Che?
> 
> That’s correct. I thought the title explains it enough. Will add more details to the description.

Thanks!
And yes, the title explains a few things, but it lacks the context (ie, the automation of a process which was done manually in the past)

File: config/operators/che/subscription.yaml (hunk 2 above)
Comment (MatousJobanek) [reply config/operators/che/subscription.yaml:24]: never mind. I don't know why, but I thought that we didn't use Che operator in the hosted toolchain :man_shrugging: 
It was either at the very beginnning or I got confused with something else :man_facepalming:

File: config/operators/pipelines/subscription.yaml (hunk 1 above)
Comment (MatousJobanek) [reply config/operators/pipelines/subscription.yaml]: May I ask you for moving it back to v0.11.0? 
I made the OSP operator running for this latest version in our member cluster, so I'm working on migrating our pipelines to the `v1beta1` api

Comment (alexeykazakov) [reply config/operators/pipelines/subscription.yaml]: Done in https://github.com/codeready-toolchain/toolchain-infra/pull/12/commits/aac812e4b608ec84aaeb0343b662c7f597b97cc8

**Comment by alexeykazakov** [comment]
@xcoulon I've added a description.

**Review by xcoulon — APPROVED** [review]
I like how the `setup_tools_operators` func is structured 🙌

**Comment by alexeykazakov** [comment]
Thanks @xcoulon !

**Pull request merged**